core/src/main/java/com/radixdlt/api/system/health/HealthInfoServiceImpl.java

```diff
@@ -106,4 +122,46 @@ public RecentSelfProposalMissStatistic recentSelfProposalMissStatistic() {
   public ProtocolState protocolState() {
     return inMemorySystemInfo.getProtocolState();
   }
+
+  @Override
+  public HealthResponse.ReadinessSignalStatusEnum readinessSignalStatus() {
+    return this.cachedReadinessSignalStatus;
+  }
+
+  @Override
+  public EventProcessor<HealthReadinessSignalStatusUpdateTrigger>
+      readinessSignalStatusUpdateTriggerEventProcessor() {
```

Comment:
maybe I don't get the full motivation here, but... why the `{trigger(10s) + processor(read DB) + read from cache on request}`, instead of a simple `read DB on request`?

Reply:
I kinda agree with Jakub, this feels a little like overkill? Out of interest, what was the push to do it this way?

Reply:
just to avoid locking on health calls

Reply:
hmh but maybe that's not really an issue if I use `access_non_locked_historical` as you suggest below

Reply:
yup, I suggest `access_non_locked_historical` for now.
Later we will read it from a DB snapshot.

We have far too many DB-reading endpoints already to care about such short-lived lock.
If you want to be super-careful (due to the public nature of /health endpoint), you can wrap it in a simple Guava cache (e.g. expire 10s after read). It would lock less often than the current solution.

Reply:
(done)